compiler/passes/codegen.cpp

```diff
@@ -806,7 +806,128 @@ static void protectNameFromC(Symbol* sym) {
 
 
 // TODO: Split this into a number of smaller routines.<hilde>
-static void codegen_header(bool isHeader) {
+static void codegen_defn() {
+  GenInfo* info = gGenInfo;
+  Vec<const char*> cnames;
+  Vec<TypeSymbol*> types;
+  std::vector<FnSymbol*> functions;
+  Vec<VarSymbol*> globals;
+
+  // reserved symbol names that require renaming to compile
+#include "reservedSymbolNames.h"
+
+  //
+  // collect types and apply canonical sort
+  //
+  forv_Vec(TypeSymbol, ts, gTypeSymbols) {
+    if (ts->defPoint->parentExpr != rootModule->block) {
+      types.add(ts);
+    }
+  }
+  qsort(types.v, types.n, sizeof(types.v[0]), compareSymbol);
+
+  //
+  // collect globals and apply canonical sort
+  //
+  forv_Vec(VarSymbol, var, gVarSymbols) {
+    if (var->defPoint->parentExpr != rootModule->block &&
+        toModuleSymbol(var->defPoint->parentSymbol)) {
+      globals.add(var);
+    }
+  }
+  qsort(globals.v, globals.n, sizeof(globals.v[0]), compareSymbol);
+
+  //
+  // collect functions and apply canonical sort
+  //
+  forv_Vec(FnSymbol, fn, gFnSymbols) {
+    functions.push_back(fn);
+  }
+  std::sort(functions.begin(), functions.end(), compareSymbol2);
+
+  FILE* hdrfile = info->cfile;
+
+  genClassIDs(types, false);
+
+  genComment("Function Pointer Table");
+//#ifndef HAVE_LLVM
+    for_vector(FnSymbol, fn2, functions) {
+      if (fn2->hasFlag(FLAG_BEGIN_BLOCK) ||
+          fn2->hasFlag(FLAG_COBEGIN_OR_COFORALL_BLOCK) ||
+          fn2->hasFlag(FLAG_ON_BLOCK)) {
+      ftableVec.add(fn2);
+      ftableMap[fn2] = ftableVec.n-1;
+      }
+    }
+  genFtable(ftableVec, false);
+//#endif
+
+  genComment("Virtual Method Table");
+  genVirtualMethodTable(types, false);
+
+  if(fIncrementalCompilation) {
+    genComment("Global Variables");
+    forv_Vec(VarSymbol, varSymbol, globals) {
+      varSymbol->codegenGlobalDef(false);
+    }
+  }
+
+  zlineToFileIfNeeded(rootModule, info->cfile);
+
+  flushStatements();
```

Comment:
It looks like the order of this line and the previous one were swapped in this change.  I have no problem if it was intentional, just wanted to mention in case it wasn't.